### PR TITLE
:recycle: Refactor systems to use Zipper

### PR DESCRIPTION
**Pull request by C0Florent**
This PR makes good use of the `Zipper` class to simplify the implementation of all the systems.

### CODE REVIEW

File: plugins/systems/collision/Collision.cpp [hunk 1]
```diff
@@ -58,44 +59,38 @@ Systems::ISystem *entryConfig(libconfig::Setting &config)
     return new Systems::Collision(config);
 }
 
-void Systems::Collision::checkPlayerProjectileToEnemyCollision(ECS::Registry &reg) {
-    auto &typeComponents = reg.componentManager().getComponents<Components::Type>();
-    auto &posComponents = reg.componentManager().getComponents<Components::Position>();
-    auto &colliderComponents = reg.componentManager().getComponents<Components::Collider>();
-    auto &damageComponents = reg.componentManager().getComponents<Components::Damage>();
-    auto &healthComponents = reg.componentManager().getComponents<Components::Health>();
+void Systems::Collision::checkPlayerProjectileToEnemyCollision(ECS::Registry &reg)
```
Comment (Tugduoff) [inline plugins/systems/collision/Collision.cpp:62]: Shouldnt the change of this file be done with the checkEnemyProjectileToPlayerCollision ?

File: src/ECS/utilities/Zipper/GenericZipper.hpp [hunk 2]
```diff
@@ -8,9 +8,41 @@
 #ifndef GENERIC_ZIPPER_HPP
     #define GENERIC_ZIPPER_HPP
 
-    #include <algorithm>
     #include <cstddef>
-    #include <vector>
+    #include <tuple>
+    #include <utility>
+
+template<typename T>
```
Comment (Tugduoff) [inline src/ECS/utilities/Zipper/GenericZipper.hpp:15]: Could you add documentation to the following functions ? So that we know what they do ?

**Review by Tugduoff — CHANGES_REQUESTED** [review]
This change makes the code way clearer, one more change could be done on one particular file and some comments on the newly created functinos could be added ! :)

File: src/ECS/utilities/Zipper/GenericZipper.hpp (hunk 2 above)
Comment (C0Florent) [reply src/ECS/utilities/Zipper/GenericZipper.hpp:15]: Sure, I intended to do it but later forgot about it

File: plugins/systems/collision/Collision.cpp (hunk 1 above)
Comment (C0Florent) [reply plugins/systems/collision/Collision.cpp:62]: Of course it should, I just didn't even see that there was another function below this one

**Review by Tugduoff — APPROVED** [review]
Seems all good ! Good job

**Review by ClementBaziret — APPROVED** [review]
Seems good to me :saluting_face:

**Pull request merged**